risc0/zkvm/src/guest/env.rs

```diff
@@ -158,51 +163,24 @@ impl Env {
     }
 
     fn commit<T: Serialize>(&mut self, data: &T) {
+        // NOTE TO REVIEWERS: This removes the send to the STDOUT channel. It will break
+        // anybody who what relying on that. It's possible send the data to
+        // STDOUT in chunks from the commit hasher if this is important.
```

Comment:
I do feel like there are some examples that use this feature, at least the battleship example was using it (IIRC).

Reply:
So a challenge with this is going to be that in order to use a fixed buffer, we can only provide a "chunked" view of the output, sending at most 512-bits at a times to the STDOUT channel. I imagine this would be surprising to users who are expecting a single call to `env::commit` to yield a single message.

What do you think about having it be the case that `env:write` works as it does now, serialized the message into one contiguous buffer and sending it in one message to `STDOUT`, but have `env::commit` no longer send to `STDOUT`? In this case, folks that need to receive data over the `STDOUT` channel will use `env::write` possibly sending the data both to commit and output.

Reply:
(Oh course, if this is a key feature, I can work on having the chunks reassembled by the host as to given the client a contiguous buffer for each message)